src/librustc_resolve/lib.rs

```diff
@@ -148,7 +148,7 @@ enum ResolutionError<'a> {
     /// error E0424: `self` is not available in a static method
     SelfNotAvailableInStaticMethod,
     /// error E0425: unresolved name
-    UnresolvedName(&'a str, &'a str, UnresolvedNameContext<'a>),
+    UnresolvedName(&'a str, &'a str, UnresolvedNameContext<'a>, bool, bool),
```

Comment:
You should either:
 * document what each of these `bool`'s represents in a comment above this variant. (I know I can infer it by grepping for the other uses of `UnresolvedName`, but I don't like having to do that to make sense of this definition), OR
 * alternative 1: use a struct-style variant (i.e. named fields) instead of a tuple-style variant here, so that you can name each piece of state
 * alternative 2: encode the significance of each `bool` here in the type.
    * (there are many ways to do this, such as introducing two new two variant enums, or introducing two structs that each just wrap around `bool`, so that the name of the struct dictates its significance.)